x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -89,6 +91,19 @@ Collection<Failure> verify(LogicalPlan plan) {
             else if (p.resolved()) {
                 return;
             }
+            // handle aggregate first to disambiguate between missing fields or incorrect function declaration
+            if (p instanceof Aggregate aggregate) {
+                for (NamedExpression agg : aggregate.aggregates()) {
```

Comment:
Should we maybe consolidate this with the verification happening below in lines 125/140? Now we're verifying aggs in two places.

Reply:
There are different types of verifications - the first one is around unknown fields the other around resolved fields but incorrectly declared.